visidata/aggregators.py

```diff
@@ -142,11 +143,39 @@ def __init__(self, pct, helpstr=''):
     def aggregate(self, col, rows):
         return _percentile(sorted(col.getValues(rows)), self.pct/100, key=float)
 
+class RankAggregator(Aggregator):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def aggregate(self, col, rows):
+        if not col.sheet.keyCols:
+            vd.error('ranking requires one or more key columns')
+            return None
+        return rank(col.sheet, rows)
 
 def quantiles(q, helpstr):
     return [PercentileAggregator(round(100*i/q), helpstr) for i in range(1, q)]
 
 
+def rank(sheet, rows):
+    def _key_progress(prog):
+        def identity(val):
+            prog.addProgress(1)
+            return val
+        return identity
+
+    keys = map(sheet.rowkey, rows)
+    with Progress(gerund='grouping', total=sheet.nRows) as prog:
+        keys_sorted = sorted(((rowkey, i) for i, rowkey in enumerate(keys)), key=_key_progress(prog))
+    # group elements by rowkey
+    with Progress(gerund='ranking', total=sheet.nRows) as prog:
```

Comment:
Using these Progress objects separately in serial will reset the progress meter.  You only need one of them, with a `total=3*sheet.nRows` (since there are 3 steps).

If you want to keep the gerunds to indicate the various steps, you can have the first one be 'outermost', and then then other Progress() within that scope with different gerunds (and no total), and only addProgress on the outer one.